src/EPR.RegulatorService.Facade.Core/Services/Producer/ProducerService.cs

```diff
@@ -28,24 +29,33 @@ public async Task<HttpResponseMessage> GetOrganisationsBySearchTerm(Guid userId,
 
         _logger.LogInformation("Attempting to fetch organisations by searchTerm '{searchTerm}'", searchTerm);
 
-        return await _httpClient.GetAsync(url);
+        var uriBuilder = new UriBuilder(_httpClient.BaseAddress)
+        {
+            Path = url
+        };
+
+        return await _httpClient.GetAsync(uriBuilder.Path);
     }
     
     public async Task<HttpResponseMessage> GetOrganisationDetails(Guid userId, Guid externalId)
     {
         var url = string.Format($"{_config.Endpoints.GetOrganisationDetails}", userId, externalId);
 
-        _logger.LogInformation("Attempting to fetch organisation details for organisation'{externalId}'", externalId);
+        _logger.LogInformation("Attempting to fetch organisation details for organisation'{ExternalId}'", externalId);
 
-        return await _httpClient.GetAsync(url);
+        var uriBuilder = new UriBuilder(_httpClient.BaseAddress)
+        {
+            Path = url
+        };
+
+        return await _httpClient.GetAsync(uriBuilder.Path);
```

Comment:
Have you tested url coming correct like previously without replacing "?" to "%3f". Please use uriBuilder.Uri.LocalPath wherever required.